kernel/fablib/runlevel/3_distribution/rsync/rsync.go

```diff
@@ -30,14 +30,24 @@ func Rsync() model.DistributionStage {
 
 func (rsync *rsyncStage) Distribute(run model.Run) error {
 	m := run.GetModel()
+	var tasks []fablib.Task
 	for regionId, r := range m.Regions {
 		for hostId, host := range r.Hosts {
 			config := newConfig(m, host.PublicIp)
-			if err := synchronizeHost(config); err != nil {
-				return fmt.Errorf("error synchronizing host [%s/%s] (%s)", regionId, hostId, err)
-			}
+			boundRegionId := regionId
+			boundHostId := hostId
+
+			tasks = append(tasks, func() error {
+				if err := synchronizeHost(config); err != nil {
+					return fmt.Errorf("error synchronizing host [%s/%s] (%s)", boundRegionId, boundHostId, err)
+				}
+				return nil
+			})
 		}
 	}
+
+	return fablib.InParallel(tasks...)
+
```

Comment:
We almost definitely don't want to do `rsync` in parallel, if that's what this is doing? At the _very least_ it needs to have configurable parallelism.

Reply:
Yes, that's what it is doing. I'm experimenting with it in a few different places. I agree, it needs to be configurable. It probably makes things slightly slower, or at least no faster, on the initial setup because you're maxing out bandwidth. On subsequent runs it probably goes a bit faster because more time is spent on establishing connections. On a refresh run my time went from 60 seconds to 50 seconds. I think most of that was from parallelizing log directory creation, not rsync.